candle-nn/src/ops.rs

```diff
@@ -44,6 +44,12 @@ pub fn sigmoid(xs: &Tensor) -> Result<Tensor> {
     (xs.neg()?.exp()? + 1.0)?.recip()
 }
 
+pub fn hard_sigmoid(xs: &Tensor) -> Result<Tensor> {
+    // TODO: Should we have a specialized op for this?
+    let six = Tensor::new(6f32, xs.device())?;
+    ((xs + 3.0) / six)?.clamp(0f32, 1f32)
```

Comment:
I think this is hard-swish and not hard-sigmoid in the pytorch nomenclature.

Reply:
I add two activation, hard-sigmoid and hard-swish.
hard-sigmoid = relu6(x+3) / 6
hard-swish = x * hard-sigmoid

Reply:
Oops, my bad, this is hard-sigmoid, can you replace the `six` with an actual `6`?

Reply:
There will be an error:
error[E0277]: cannot divide `std::result::Result<candle_core::Tensor, candle_core::Error>` by `{float}`
  --> candle-nn/src/ops.rs:50:17
   |
50 |     ((xs + 3.0) / 6.0)?.clamp(0f32, 1f32)
   |                 ^ no implementation for `std::result::Result<candle_core::Tensor, candle_core::Error> / {float}`
   |
   = help: the trait `std::ops::Div<{float}>` is not implemented for `std::result::Result<candle_core::Tensor, candle_core::Error>`
   = help: the following other types implement trait `std::ops::Div<Rhs>`:
             <std::result::Result<B, candle_core::Error> as std::ops::Div<candle_core::Tensor>>
             <std::result::Result<B, candle_core::Error> as std::ops::Div<&candle_core::Tensor>>

For more information about this error, try `rustc --explain E0277`.

Reply:
Try `((xs + 3.0)? / 6)?` instead?

Reply:
`((xs + 3.0)? / 6)?.clamp(0f32, 1f32)` still got an error:     
error[E0277]: the trait bound `{integer}: std::borrow::Borrow<candle_core::Tensor>` is not satisfied
  --> candle-nn/src/ops.rs:53:18
   |
53 |     ((xs + 3.0)? / 6)?.clamp(0f32, 1f32)
   |                  ^ the trait `std::borrow::Borrow<candle_core::Tensor>` is not implemented for `{integer}`
   |
   = note: required for `candle_core::Tensor` to implement `std::ops::Div<{integer}>`

For more information about this error, try `rustc --explain E0277`.
warning: `candle-nn` (lib) generated 1 warning
error: could not compile `candle-nn` (lib) due to previous error; 1 warning emitted

But, This will be Ok

Reply:
Yes, should be 6.0 instead of 6 indeed. Please use `div` instead of `/` and should be all good.

Reply:
Ok, fixed it.